pkg/genericapiserver/server/openapi/openapi.go

```diff
@@ -79,6 +81,15 @@ func (o *openAPI) init(webServices []*restful.WebService) error {
 			return r.Operation, nil, nil
 		}
 	}
+	if o.config.GetDefinitionName == nil {
+		o.config.GetDefinitionName = func(_, name string) (string, []string) {
+			return name[strings.LastIndex(name, "/")+1:], nil
+		}
+	}
+	o.definitions = o.config.GetDefinitions(func(name string) spec.Ref {
+		defName, _ := o.config.GetDefinitionName(o.servePath, name)
+		return spec.MustCreateRef("#/definitions/" + openapi.EscapeJsonPointer(defName))
```

Comment:
The escaping is a standard which without it, the spec would be invalid (the validation will fail for it). The client and doc generators are already doing some preprocessing before consuming this spec, and even without this escaping we need to do more preprocessing to use this new spec. Adding a step to deal with this in the preprocessing shouldn't be hard (if the standard json readers does not deal with it).